### ISSUE TITLE
VEL.ROT always zero when using Python API

### ISSUE THREAD

**Issue by uahic** [deepmind/lab#180]
Related to: https://github.com/deepmind/lab/issues/140

It seems that the VEL.ROT output (at least true for the Python API) stays (0., 0., 0.) for all actions that I've tried out. 

Example: repeated actions like 'LOOK_LEFT_RIGHT_PIXELS_PER_FRAME' with an amount of   228 (int)

Reproducing the issue:
```
import numpy as np
import deepmind_lab

level = 'tests/empty_room_test'
img_width = 64
img_height = 64
observations = ['RGB',
                'DEBUG.POS.ROT', 'DEBUG.POS.TRANS',
                'VEL.ROT', 'VEL.TRANS']
lab_config = {'width': str(img_width), 'height': str(
    img_height), 'fps': '30'}

env = deepmind_lab.Lab(level, observations,
                       config=lab_config, renderer='hardware')
env.reset()

for _ in range(100):
    action = np.zeros([len(env.action_spec())], dtype=np.intc)
    action[0] = 228
    env.step(action)
    print(env.observations()['VEL.ROT'])
```

Notice, however when playing manually DM_LAB, it works fine. May it happen that VEL.ROT is only calculated for mouse inputs?

Command for that:
```
bazel run :game -- --level_script=tests/empty_room_test --observation VEL.ROT

```

**Comment by tkoeppe** [comment]
Thanks for the report! The code in question is this: https://github.com/deepmind/lab/blob/master/deepmind/engine/context_game.cc#L466-L482

The incoming new angles always seem to be the same as the existing ones, so the velocity is always zero, but I don't know why. @charlesbeattie?

**Comment by tkoeppe** [comment]
Charlie found a lead for this (the SetPlayerState update is called repeatedly at the same time point during rendering); we're working on a fix.

**Comment by tkoeppe** [comment]
(We think the fix will look like this:

```patch
--- deepmind/engine/context_game.cc
+++ deepmind/engine/context_game.cc
@@ -473,6 +473,10 @@
                                  int team_score, int other_team_score,
                                  int player_id, bool teleporter_flip,
                                  int timestamp_msec) {
+  // Ignore repeated calls at the same timestamp.
+  if (timestamp_msec == player_view_.timestamp_msec) {
+    return;
+  }
   PlayerView before = player_view_;
   std::copy_n(pos, 3, player_view_.pos.data());
   std::copy_n(vel, 3, player_view_.vel.data());
```

I'll apply that soon.)

**Comment by tkoeppe** [comment]
This should be fixed by b30c98bfdaf21cb3ae10afdde1e2b1013f020e91.
